src/components/dialog/dialog.js

```diff
@@ -1,107 +1,293 @@
+     */
+    function showDialog(dialog) {
+        //  Focus the modal and remove aria attributes
+        dialog.setAttribute('tabindex', 1);
+        dialog.setAttribute('aria-hidden', false);
+
+        //  Set the first and last focusable elements
+        state.focusableElements = qa(NATIVELY_FOCUSABLE_ELEMENTS.join(), dialog);
+
+        //  focus first element if exists, otherwise focus dialog element
+        if (state.focusableElements.length) {
+            state.focusableElements[0].focus();
+        } else {
+            dialog.focus();
+        }
+
+        //  Bind events
+        defer(bindKeyCodeEvents);
+        defer(bindCloseEvents);
+        if (!isModal && DOM.backdrop) {
+            defer(bindBackdropEvents);
+        }
+
+        // Add backdrop if needed
+        if (DOM.backdrop) {
+            DOM.page.appendChild(DOM.backdrop);
         }
 
-        // Set page attribute
-        DOM.page.setAttribute('data-ui-dialog', 'is-initialised');
+        //  Add class to make dialog visible
+        dialog.classList.add(activeClass);
+    }
+
+
+    /**
+     * @function bindCloseEvents
+     * @desc Finds all close buttons and attaches click event listener
+     * @param {node} dialog
+     */
+    function bindCloseEvents(dialog = state.currentDialog) {
+        // Grab all buttons which open this instance of the modal
+        const closeButtons = qa(closeBtn);
+
+        closeButtons.forEach(button => button.addEventListener('click', closeDialog));
+    }
 
-        // Find dialog and hide if not already hidden
-        DOM.dialog.classList.add(defaultClassNames.IS_HIDDEN);
 
-        // Attach event listeners
-        DOM.openBtn.addEventListener('click', show, false);
-        DOM.closeBtn.addEventListener('click', hide, false);
-        if (!isModal) {
-            backdrop.addEventListener('click', hide, false);
+    /**
+     * @function closeDialog
+     * @desc Sets up dialog ready to be hidden
+     */
+    function closeDialog() {
+        hideDialog(state.currentDialog);
+    }
+
+
+    /**
+     * @function hideDialog
+     * @desc adds aria attributes and hides dialog, removing backdrop if needed
+     * @param {node} dialog
+     */
+    function hideDialog(dialog) {
+
+        //  show container and focus the dialog
+        dialog.setAttribute('aria-hidden', true);
+        dialog.removeAttribute('tabindex');
+
+        // Unbind events
+        unbindKeyCodeEvents();
+        if (!isModal && DOM.backdrop) {
+            unbindBackdropEvents();
+        }
+
+        //  Remove active state hook class
+        dialog.classList.remove(activeClass);
+
+        // Remove backdrop if needed
+        if (DOM.backdrop) {
+            DOM.page.removeChild(DOM.backdrop);
         }
-        document.addEventListener('keydown', keyHandler, false);
+
+        // Reset state and return focus to button that opened the dialog
+        state.currentOpenButton.focus();
```

Comment:
As per previous comment let's discuss best approach for state / instance dom nodes vs page / constant nodes